### ISSUE TITLE
fixMantissaOverflow should not round up

### ISSUE THREAD

**Issue by IvanSanchez** [antimatter-dimensions/notations#123]
The current implementation of `fixMantissaOverflow` seems to always round up (when the first non-significant digit is 5), as noted on https://github.com/antimatter-dimensions/notations/blob/37990325b8d055493ac05ed2259366dbbe5cfe3c/spec/infix/infix-eng.spec.js#L14-L15

I think it'd be better to implement the ["round half to even"](https://en.wikipedia.org/wiki/Rounding#Round_half_to_even) algorithm (see also the [IEEE754 stuff about rounding](https://en.wikipedia.org/wiki/IEEE_754#Rounding_rules)).

In a nutshell:

> A tie-breaking rule without positive/negative bias and without bias toward/away from zero is round half to even. By this convention, if the fractional part of x is 0.5, then y is the even integer nearest to x. Thus, for example, +23.5 becomes +24, as does +24.5; while −23.5 becomes −24, as does −24.5. This function minimizes the expected error when summing over rounded figures, even when the inputs are mostly positive or mostly negative.

**Comment by dan-simon** [comment]
`fixMantissaOverflow` isn't doing rounding in those cases, the rounding up is because 3.45 and 3.35 are rounded to one decimal place as 3.4 and 3.3 (line 58 of community/infix-abstract.ts, specifically `value.mantissa.toFixed(places)`). For example, 35.5 is rounded to 35, which is not rounding up, but can be explained by `(3.55).toFixed(1)` being 3.5.

The default formatUnder1000 (used by, for example, Scientific notation) also has this issue. This is because `x.toFixed(0)` in JS uses round half away from zero rather than round half to even. I'm not sure how to generalize `x.toFixed(i)` for i not equal to zero to use round half to even, but my best quickly thought out guess is something equivalent to `if (4 * Math.pow(10, i) * x % 4 === 1) {return (x - Math.pow(10, -i) / 2).toFixed(i)} else if (4 * Math.pow(10, i) * x % 4 === 3) {return (x + Math.pow(10, -i) / 2).toFixed(i)} else {return x.toFixed(i)}`. However, this might have floating point issues.
